pkg/controller/cert_control.go

```diff
@@ -0,0 +1,323 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	certutil "github.com/pingcap/tidb-operator/pkg/util/crypto"
+	capi "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	types "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CertControlInterface manages certificates used by TiDB clusters
+type CertControlInterface interface {
+	Create(ns string, instance string, commonName string, hostList []string, IPList []string, component string, suffix string) error
```

Comment:
Pass too many parameters，I think we can pass `*v1alpha1.TidbCluster` instead of `ns` and `instance`，and define a `tidbClusterCert` struct to store certificate related content，like [kubeadm](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/certs/certlist.go#L34-L43) does.